### PR TITLE
[4.0] Admin menu redundant link

### PR DESCRIPTION
**Pull request by brianteeman**
When adjacent links go to the same location this results in additional navigation and repetition for keyboard and screen reader users. WCAG 2.4.4

The parent menu item for Smart Search is the same as the first child  menu item.

Although the first one is disabled by the js and only acts as means to expand the submenu its still a problem for some assistive technologies.

This PR follows the pattern of all other core components and changes the parent link to index.php?option=com_finder

This can be tested either by performing a clean installation or by manually applying the sql in the update script.

Note that "database fix" will not apply the change in the update script as it is a content change not a structural change.

### before
image

### after
image

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 2e49b38e2ccea79159016102aa91040c47c9ceee<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30696">issues.joomla.org/tracker/joomla-cms/30696</a>.</sub>

**Comment by infograf768** [comment]
I am not sure about that
`UPDATE `#__menu` SET `link`='index.php?option=com_finder' WHERE `id`=13;`

default com_finder menu in 3.x is not at id 13 but 18, which may mean that an update would chnage com_newsfeeds link instead of com_finder, if I do not mistake.

**Comment by brianteeman** [comment]
That is a different issue if correct

**Comment by richard67** [comment]
No, it is not a different issue. You have to change the `WHERE` clause in your update SQL script to not use the ID, except if you can be sure that on any installation with any long update history it always has the same ID. This can not be granted, and so the golden rule is to never use the ID to identify a menu item for an update with an SQL update script.

If you want I can advise with a proper `WHERE`clause.

**Comment by richard67** [comment]
I suggest to use the following where clause:
```
WHERE `menutype`='main' AND `title`='com_finder' AND `link`='index.php?option=com_finder&view=index'
```

**Comment by richard67** [comment]
For example if someone had a Joomla 1.5 or so installation and updated it to 2.5, then the menuitem ID would be 21, see update SQL `2.5.0-2011-12-22.sql` in the `Joomla_2.5.28-Stable-Full_Package.zip`, and it will still be 21 now after the years and all updates.

**Comment by alikon** [comment]
it seems more correct imho, the only left question is about the alternative admin menus that can be configured .....

**Comment by richard67** [comment]
> it seems more correct imho, the only left question is about the alternative admin menus that can be configured .....

@alikon Those will not have `menutype`='main'.

And no, it is not more correct, it is just correct, because using the ID as criteria in the WHERE clause is DEFINITELY wrong, see my previous comment about ID of that menu item being 21 when coming from pre-2.5 in the update history, and as @infograf768 wrote it is 18 on a new installed staging.

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-09-19.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+UPDATE `#__menu` SET `link`='index.php?option=com_finder' WHERE `id`=13;
```
Comment (richard67) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-09-19.sql]:
```suggestion
UPDATE `#__menu` SET `link`='index.php?option=com_finder' WHERE `menutype`='main' AND `title`='com_finder' AND `link`='index.php?option=com_finder&view=index';
```
Just to save you some work so you can apply it with one click.

File: administrator/components/com_admin/sql/updates/postgresql/4.0.0-2020-09-19.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+UPDATE "#__menu" SET "link"='index.php?option=com_finder' WHERE "id"=13;
```
Comment (richard67) [inline administrator/components/com_admin/sql/updates/postgresql/4.0.0-2020-09-19.sql]:
```suggestion
UPDATE "#__menu" SET "link"='index.php?option=com_finder' WHERE "menutype"='main' AND "title"='com_finder' AND "link"='index.php?option=com_finder&view=index';
```
Just to save you some work so you can apply it with one click.

**Comment by alikon** [comment]
sorry my bad .... i've not checked deep that alternative admin menu have a proper different `menutype`

**Comment by brianteeman** [comment]
Happy to update - you should note that the code was copied from
administrator\components\com_admin\sql\updates\mysql\4.0.0-2019-05-05.sql

**Comment by richard67** [comment]
> you should note that the code was copied from
> administrator\components\com_admin\sql\updates\mysql\4.0.0-2019-05-05.sql

@brianteeman OMG!!! I have to check and if possible fix that! Thank you for that information,

**Comment by alikon** [comment]
oh shit ..... i need to ask to myself where i was when that s..... was proposed/merged

**Comment by richard67** [comment]
@alikon Initially it was that one: #24801 . But let's continue that discussion somewhere else in order not to pollute this PR with off-topic. I will check and if necessary make an issue or PR.

**Comment by alikon** [comment]
sure right......ping me if you need help/tests on that matter

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on f579ff9cb6828ff1a972ccd9698833891737a844<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30696">issues.joomla.org/tracker/joomla-cms/30696</a>.</sub>

**Comment by richard67** [comment]
@alikon Did you do a real test? Or just code review? Am preparing real test, but it takes time. It needs to test new installation with this PR applied and also update SQL statements test on a 4.0-dev for both MySQL and PostgreSQL.

**Comment by alikon** [comment]
i've made a quick test on my current staging 4.0-dev.... probably this is not enough....well i'll remove my test

**Comment by alikon** [comment]
I have not tested this item.<br /><br />too much quick test<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30696">issues.joomla.org/tracker/joomla-cms/30696</a>.</sub>

**Comment by richard67** [comment]
@alikon No, it's ok. I'll do the complete test. From you it's ok with 1 real test and a code review.

**Comment by richard67** [comment]
@alikon In your case, as you are an SQL expert with both MySQL and PostgreSQL, a code review could be sufficient ;-) Just wanted to know how you had tested, that's why I had asked. If you had said you have done all real tests I could have decided to be lazy and do the code review only ;-)

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on f579ff9cb6828ff1a972ccd9698833891737a844<br /><br />Tested
- New installation with patch applied on MySQL
- Update SQL applied on clean 4.0-dev on MySQL
- New installation with patch applied on PostgreSQL
- Update SQL applied on clean 4.0-dev on PostgreSQL<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30696">issues.joomla.org/tracker/joomla-cms/30696</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on f579ff9cb6828ff1a972ccd9698833891737a844<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30696">issues.joomla.org/tracker/joomla-cms/30696</a>.</sub>

**Comment by infograf768** [comment]
rtc
happy i could help...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30696">issues.joomla.org/tracker/joomla-cms/30696</a>.</sub>

**Comment by infograf768** [comment]
Tks

**Comment by brianteeman** [comment]
thanks

**Pull request merged**